sigstore/_internal/rekor/_client.py

```diff
@@ -43,6 +43,23 @@ def from_response(cls, dict_) -> RekorEntry:
         )
 
 
+@dataclass(frozen=True)
```

Comment:
Nitpick: We should consider using Pydantic's `BaseModel` + `Field` here, so that we can do simplify to this:

```python
class RekorInclusionProof(BaseModel):
    log_index: int = Field(..., alias="logIndex")
    # other fields
```

That removes the need for the custom `from_dict` constructor + eliminates a potential source of errors via discrepancies.

Reply:
This is tiny, so let's postpone.

Reply:
No worries. Did this in [ac03fd1](https://github.com/trailofbits/sigstore-python/pull/26/commits/ac03fd197b6324304c42713eb09d9e27192016a2).